### PR TITLE
Correct pylance errors; check for correct extension

### PR DESCRIPTION
**Pull request by nekoffski**

### CODE REVIEW

File: src/foxlator_lib/video.py [hunk 1]
```diff
@@ -9,12 +11,24 @@ class VideoError(BaseError):
...
     def get_audio_path(self) -> AudioPath:
         return AudioPath()
+
+    def _validate_extension(self, path: str):
+        extension: str = pathlib.Path(path).suffix
+        if len(extension) == 0:
+            raise VideoError(
+                "Provided file path is invalid or doesn't contain extension")
+        if extension not in self.known_extensions:
+            raise VideoError(f"Unsupported file extension: {extension}")
```
Comment (witekha3) [inline src/foxlator_lib/video.py:34]: The "extention" variable will contain values with a dot prefix, such as ".mp4". For this reason, this method will always return an error. There is onlo small nit from my side. I don't think we need a separate error for the "wrong" extension, we can simply return one message, as in the example above (with the correction)

```
    def _validate_extension(self, path: str):
        extension: str = pathlib.Path(path).suffix
        extension = extension.split(".")[1] if "." in extension else extension
        if extension not in self.known_extensions or not extension:
            raise VideoError(f"Provided file path is invalid or doesn't contain extension: {path}")
```

Or just add "." to values in known_extensions

Comment (nekoffski) [reply src/foxlator_lib/video.py:34]: Good catch. I'll fix that. For the second part, I would keep it as those are separated cases - lack of extension is not the same as providing extension that is not supported.

Comment (nekoffski) [reply src/foxlator_lib/video.py:34]: Corrected.

File: test/video_tests.py
Comment (witekha3) [inline test/video_tests.py:1]: Shouldn't the test files end with "_test.py"?

Comment (nekoffski) [reply test/video_tests.py:1]: why not tests?

Comment (nekoffski) [reply test/video_tests.py:1]: There are multiple of them in a single file :P

File: test/utils/base.py [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+    def __init__(self):
+        self.is_file_queue: queue.SimpleQueue[bool] = queue.SimpleQueue()
+
+    def is_file(self, path: str) -> bool:
+        if self.is_file_queue.empty():
+            return False  # False by default
+        return self.is_file_queue.get()
+
+    def push_is_file_return(self, value: bool):
+        self.is_file_queue.put(value)
```
Comment (witekha3) [inline test/utils/base.py]: Shouldn't we mock it for example like:

`with patch('os.path.isfile', return_value=True):`

Or via decorator?

File: test/video_tests.py
Comment (witekha3) [inline test/video_tests.py:1]: A test showing "success" is missing

**Review by witekha3 — CHANGES_REQUESTED** [review]

File: test/video_tests.py
Comment (nekoffski) [reply test/video_tests.py:1]: It's not yet implemented as the class is not finished yet.

Comment (witekha3) [reply test/video_tests.py:1]: Although the files can contain many tests, they are usually called "test_*.py" or "*_test.py". I did a little research and again, other projects use this convention. This convention can also be found in the examples at docs.python.org

File: test/utils/base.py (hunk 2 above)
Comment (witekha3) [reply test/utils/base.py]: @nekoffski what do you think about that?

**Review by witekha3 — CHANGES_REQUESTED** [review]

File: test/video_tests.py
Comment (nekoffski) [reply test/video_tests.py:1]: Not sure if that matters at all but as you wish.

File: test/utils/base.py (hunk 2 above)
Comment (nekoffski) [reply test/utils/base.py]: Sure, let's use mock library.

**Review by witekha3 — APPROVED** [review]
Small nit: I would use patch as a decorator. Besides that lgtm

**Pull request merged**